plugins/aws/fix_plugin_aws/resource/cloudwatch.py

```diff
@@ -364,11 +365,75 @@ class AwsCloudwatchLogGroup(LogsTaggable, AwsResource):
     group_metric_filter_count: Optional[int] = field(default=None, metadata=dict(ignore_history=True))
     group_stored_bytes: Optional[int] = field(default=None, metadata=dict(ignore_history=True))
     group_data_protection_status: Optional[str] = field(default=None)
+    group_policy: Optional[Json] = field(default=None)
 
     def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
         if kms_key_id := source.get("kmsKeyId"):
             builder.dependant_node(self, clazz=AwsKmsKey, id=AwsKmsKey.normalise_id(kms_key_id))
 
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def add_log_group_policy(group: AwsCloudwatchLogGroup) -> None:
+            def is_arn_match(resource_arn: str, target_arn: str) -> bool:
+                if resource_arn == target_arn:
+                    return True
+                if resource_arn.endswith(":*"):
+                    return target_arn.startswith(resource_arn[:-1])
+                return False
+
+            def parse_resource_arn(resource: Any) -> List[str]:
+                if isinstance(resource, str):
+                    return [resource.split(":log-stream:")[0]]
+                elif isinstance(resource, list):
+                    return [arn.split(":log-stream:")[0] for arn in resource]
+                return []
+
+            def process_log_group_policies(raw_policies: List[Dict[str, Any]], target_group_arn: str) -> Dict[str, Any]:
+                associated_policies = {}
+                for policy in raw_policies:
+                    policy_name = policy.get("policyName", "Unknown")
+                    policy_document = json_loads(policy.get("policyDocument", "{}"))
+                    policy_statements = policy_document.get("Statement", [])
+
+                    if not isinstance(policy_statements, list):
+                        policy_statements = [policy_statements]
+
+                    for statement in policy_statements:
+                        statement_resources = statement.get("Resource")
+                        log_group_arns = parse_resource_arn(statement_resources)
+
+                        if any(is_arn_match(arn, target_group_arn) for arn in log_group_arns):
+                            # If a match is found, associate the policy and move to the next policy
+                            associated_policies[policy_name] = policy
+                            break
```

Comment:
@meln1k Is there something in your access edges PR that would simplify this logic?

Reply:
unfortunately not.